sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -306,7 +306,7 @@ public static bool ExtensionMethodsCalledForAllDiagnostics(IEnumerable<Diagnosti
                 // why we require each diagnostic to be reported through the extension methods at least once.
                 return analyzers
                     .SelectMany(analyzer => analyzer.SupportedDiagnostics)
-                    .Select(d => counters.GetValueOrDefault(d.Id))
+                    .Select(d => DictionaryExtensions.GetValueOrDefault(counters, d.Id))
```

Comment:
why is this change necessary?

Reply:
.Net Core provides `System.Collections.Generic.CollectionExtensions.GetValueOrDefault<TKey, TValue>(System.Collections.Generic.IReadOnlyDictionary<TKey, TValue>, TKey)'` which conflicts with `'SonarAnalyzer.Helpers.DictionaryExtensions.GetValueOrDefault<TKey, TValue>(System.Collections.Generic.IDictionary<TKey, TValue>, TKey)'`.

Since in .Net Framework the method from `System.Collections.Generic` is not available, we call `DictionaryExtensions.GetValueOrDefault` as a method instead of extension. This avoids the conflict and usage of conditional code.